pkgs/tools/networking/moodle-dl/default.nix

```diff
@@ -0,0 +1,34 @@
+{ lib, python3Packages }:
+python3Packages.buildPythonApplication rec {
+  pname = "moodle-dl";
+  version = "2.1.2.5";
+
+  src = python3Packages.fetchPypi {
+    inherit pname version;
+    sha256 = "1gc4037dwyi48h4vi0bam23rr7pfyn6jrz334radz0r6rk94y8lz";
+  };
+
+  patches = [
+    # nixpkgs doesn't have version 2.0.1, so we use 2.0.0
+    ./readchar-version.patch
```

Comment:
Please use substituteInPlace in postPatch instead.

Also can we just update readchar to 2.0.1?

Reply:
Updating readchar to 2.0.1 is nontrivial, because the 2.0.1 tag wasn't published to GitHub by its author. Thus we cannot pull the package from GitHub. PyPI would, I suppost, work, but in readchar's package file it says `Don't use wheels on PyPI`. So I guess I shouldn't get it from there.